tutorials/xarray_tutorial.py

```diff
@@ -6,295 +6,344 @@
+# ``DataArray``\s by accepting them as inputs, returning them as outputs, and automatically
+# using the attached coordinate data/metadata to determine grid arguments
+
+heights = data_parsed.metpy.parse_cf('Geopotential_height_isobaric').metpy.sel(
+    time='2017-09-05 18:00',
+    vertical=50000.
+)
+u_g, v_g = mpcalc.geostrophic_wind(heights)
+u_g
 
-heights = data['height'].metpy.loc[{'time': time[0], 'vertical': 500. * units.hPa}]
-lat, lon = xr.broadcast(y, x)
-dx, dy = grid_deltas_from_dataarray(heights)
-u_geo, v_geo = mpcalc.geostrophic_wind(heights, dx, dy, lat)
-print(u_geo)
-print(v_geo)
+#########################################################################
+# For profile-based calculations (and most remaining calculations in the ``metpy.calc``
+# module), xarray ``DataArray``\s are accepted as inputs, but the outputs remain Pint
+# Quantities (typically scalars)
+
+data_at_point = data.metpy.sel(
+    time1='2017-09-05 12:00',
+    latitude=40 * units.degrees_north,
+    longitude=260 * units.degrees_east
+)
+dewpoint = mpcalc.dewpoint_from_relative_humidity(
+    data_at_point['Temperature_isobaric'],
+    data_at_point['Relative_humidity_isobaric']
+)
+cape, cin = mpcalc.surface_based_cape_cin(
+    data_at_point['isobaric3'],
+    data_at_point['Temperature_isobaric'],
+    dewpoint
+)
+cape
 
 #########################################################################
-# Plotting
-# --------
+# Remaining portions of MetPy's calculations (such as its interpolation module) do not fully
+# support xarray, and so, use of ``.values`` may be needed to convert to a bare NumPy array.
+# For full information on xarray support for your function of interest, see the
+# `Reference Guide <../api/index.html>`_.
 #
-# Like most meteorological data, we want to be able to plot these data. DataArrays can be used
-# like normal numpy arrays in plotting code, which is the recommended process at the current
-# point in time, or we can use some of xarray's plotting functionality for quick inspection of
-# the data.
+# CF-Compliant Dataset Example
+# ----------------------------
 #
-# (More detail beyond the following can be found at `xarray's plotting reference
-# <http://xarray.pydata.org/en/stable/plotting.html>`_.)
+# The GFS sample used throughout this tutorial so far has been an example of a CF-compliant
+# dataset. These kinds of datasets are easiest to work with it MetPy, since most of the
+# "xarray magic" uses CF metadata. For this kind of dataset, a typical workflow looks like the
+# following
+
+# Load data, parse it for a CF grid mapping, and promote lat/lon data variables to coordinates
+data = xr.open_dataset(
+    get_test_data('narr_example.nc', False)
+).metpy.parse_cf().set_coords(['lat', 'lon'])
+
+# Subset to only the data you need to save on memory usage
+subset = data.metpy.sel(isobaric=500 * units.hPa)
+
+# Quantify if you plan on performing array operations that need to maintain unit correctness
```

Comment:
maybe "xarray operations"? Just want to be completely clear that `quantify()` isn't necessary for MetPy calculations.